py/client/pydeephaven/_table_ops.py

```diff
@@ -606,17 +606,21 @@ def make_grpc_request_for_batch(self, result_id, source_id) -> Any:
 
 
 class CreateInputTableOp(TableOp):
-    def __init__(self, schema: pa.schema, init_table: Any, key_cols: List[str] = None):
+    def __init__(self, schema: pa.schema, init_table: Any, key_cols: List[str] = None, blink: bool = False):
         self.schema = schema
         self.init_table = init_table
         self.key_cols = key_cols
+        self.blink = blink
 
     @classmethod
     def get_stub_func(cls, table_service_stub: table_pb2_grpc.TableServiceStub) -> Any:
         return table_service_stub.CreateInputTable
 
     def make_grpc_request(self, result_id, source_id) -> Any:
-        if self.key_cols:
+        if self.blink:
+            blink_ = table_pb2.CreateInputTableRequest.InputTableKind.Blink()
+            input_table_kind = table_pb2.CreateInputTableRequest.InputTableKind(blink=blink_)
+        elif self.key_cols:
```

Comment:
I see the check below, but there should be a check here at a minimum to ensure that these objects are only created with valid state.

Reply:
I am not sure that's needed, 1) as noted, the check already happens in `input_table` - the public interface; 2) the conditional branching here matches what the docstring of  `input_table`.
```python
    def input_table(self, schema: pa.Schema = None, init_table: Table = None,
                    key_cols: Union[str, List[str]] = None, blink_table: bool = False) -> InputTable:
        """Creates an InputTable from either Arrow schema or initial table.  When blink_table is True, the InputTable
        will be a blink table. When blink_table is False (default), the InputTable will be
        keyed if key columns are provided, otherwise it will be append-only.
```

Reply:
It may be internal, but that just means we internally screw it up if we make changes later.